src/meta/src/stream/stream_graph/fragment.rs

```diff
@@ -682,6 +696,74 @@ impl StreamFragmentGraph {
             )
         })
     }
+
+    /// Collect the mapping from table / `source_id` -> `fragment_id`
+    pub fn collect_backfill_mapping(&self) -> HashMap<u32, Vec<FragmentId>> {
+        let mut mapping = HashMap::new();
+        for (fragment_id, fragment) in &self.fragments {
+            let fragment_id = fragment_id.as_global_id();
+            let fragment_mask = fragment.fragment_type_mask;
+            let candidates = [
+                FragmentTypeFlag::CrossDbSnapshotBackfillStreamScan,
+                FragmentTypeFlag::SnapshotBackfillStreamScan,
+                FragmentTypeFlag::StreamScan,
```

Comment:
`StreamScan` should have included all `CrossDbSnapshotBackfillStreamScan` and `SnapshotBackfillStreamScan`.

Reply:
Fixed https://github.com/risingwavelabs/risingwave/pull/20967/commits/d5b053f003f71ed81dd3e552c20afe97e5ae2b6b